CodebustersPAS-ClientApp/src/pages/Teacher/InstructorDashboard.tsx

```diff
@@ -1,24 +1,67 @@
-import React from 'react';
-import { useParams } from 'react-router-dom';
-import './InstructorDashboard.css';
+import React, { useEffect } from "react";
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  useLocation,
+  useNavigate,
+} from "react-router-dom";
+import SignIn from "../LoginPage";
+import CreateAccount from "../CreateAccountPage";
+import GroupEvaluation from "../CourseEvaluation/GroupEvaluation";
+import SelfAssessment from "../CourseEvaluation/Self-Assessment";
+import PeerAssessment from "../Students/PeerAssessment";
+import TeamOverview from "../Teacher/CreateTeam/TeamOverview";
+import Navbar from "../NavBar";
+import SummaryComments from "../CourseEvaluation/Summary_Comments";
+import InstructorDashboard from "../Teacher/InstructorDashboard";
+import TeamEvaluation from "../Teacher/TeamEvaluation"; // Import the TeamEvaluation component
 
-const InstructorDashboard: React.FC = () => {
+const App: React.FC = () => {
+  const navigate = useNavigate();
 
-    const { groupName } = useParams<{ groupName: string }>();
+  useEffect(() => {
+    if (process.env.NODE_ENV === 'deploy') {  
+      const redirectPath = sessionStorage.getItem("redirectPath");
+      if (redirectPath) {
+        sessionStorage.removeItem("redirectPath");
+        console.log("Redirecting to: ", redirectPath);
+        navigate(redirectPath);
+      }
+    }
+  }, [navigate]);
+
+  const location = useLocation();
+
+  const showLogout = !["/", "/signup"].includes(location.pathname);
 
   return (
-    <div className="dashboard-container">
-      <div className="header">
-        <h1>Details for Course: {groupName} </h1>
-      </div>
-      <div className="links-container">
-        <a href="/summary" className="dashboard-link">Summary of results</a>
-        <a href="/detailed-summary" className="dashboard-link">Detailed Summary of results</a>
-        <a href="/charts" className="dashboard-link">Charts</a>
-        <a href="/Teacher/TeamOverview" className="dashboard-link">Return to main menu</a>
-      </div>
+    <div className="App">
+      <Navbar showLogout={showLogout} />
+      <div className="content"></div>
+      <Routes>
```

Comment:
Why did you add the router from main to this component?

Reply:
> Why did you add the router from app.tsx to this component?

Yeah, it shouldn't have been there. I took that part out.